dask/highlevelgraph.py

```diff
@@ -88,9 +194,23 @@ class HighLevelGraph(Mapping):
         typically used by developers to make new HighLevelGraphs
     """
 
-    def __init__(self, layers, dependencies):
+    def __init__(
+        self,
+        layers: Mapping[str, Mapping],
+        dependencies: Mapping[str, Set],
+        key_dependencies: Optional[Mapping[Hashable, Set]] = None,
+    ):
+        self.__keys = None
         self.layers = layers
         self.dependencies = dependencies
+        self.key_dependencies = key_dependencies
+
+    def keyset(self):
+        if self.__keys is None:
```

Comment:
Did you intend to use double-underscore here?  If so, are you aware of the special behavior that Python classes and subclasses have around this?  This is a rarely-used feature.

Reply:
Yes, `__keys` is used as a cache of `keyset()` and my idea was to make it easy for future sub-classes of `HighLevelGraph` to ignore. Having said that, I have to problem renaming it `self._keys`.